include/otf2xx/definition/metric_class.hpp

```diff
@@ -38,13 +38,21 @@
 #include <otf2xx/fwd.hpp>
 #include <otf2xx/reference.hpp>
 
-#include <otf2xx/definition/detail/base.hpp>
 #include <otf2xx/definition/detail/metric_class_impl.hpp>
+#include <otf2xx/definition/detail/referable_base.hpp>
 
 namespace otf2
 {
 namespace definition
 {
+    namespace detail
+    {
+        class metric_base
+        {
+        public:
+            using reference_type = otf2::reference_impl<metric_base, metric_base>;
```

Comment:
Why? Duplicates that tag type and `metric_base` and `io_file_base` shouldn't be instantiated (tags only)

BTW: https://silc.zih.tu-dresden.de/otf2-current/group__records__definition.html shows the interfaces `Metric` and `IoFile`. So if at all, I'd name those classes `metric` and `io_file`, leave them empty except the reference_type and make the other classes inherit from them (additionally) to mirror the class diagram specified by OTF2.   
But easier would be to just declare them and use them as tags only (again I prefer the OTF2 names instead of `_base` as they are no bases)